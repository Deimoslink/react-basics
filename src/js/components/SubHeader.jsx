import React from 'react';

export class SubHeader extends React.Component {
    render() {
        return (
            <div className="sub-header">
                <div>
                    <span>8 movies found</span>
                </div>
                <div className="filter-panel">
                    <span>Sort by</span>
                    <button className="button button-simple">release date</button>
                    <button className="button button-simple active">rating</button>
                </div>
            </div>
        );
    }
}
