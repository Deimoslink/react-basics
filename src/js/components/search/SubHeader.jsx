import React from 'react';
import {connect} from 'react-redux';
import {setSorting} from '../../actions';
import {getSortBy} from '../../selectors';

class SubHeader extends React.Component {
    render() {
        const {setSorting} = this.props;
        return (
            <div className="sub-header">
                <div>
                    <span hidden={this.props.total === 0}>{this.props.total} movies found</span>
                </div>
                <div className="filter-panel">
                    <span>Sort by</span>
                    <button onClick={() => setSorting('release_date')}
                            className={`button button-simple ${this.props.sortBy === 'release_date' ? 'active' : null}`}>
                        release date
                    </button>
                    <button onClick={() => setSorting('vote_average')}
                            className={`button button-simple ${this.props.sortBy === 'vote_average' ? 'active' : null}`}>
                        rating
                    </button>
                </div>
            </div>
        );
    }
}

export default connect(
    state => ({
        sortBy: getSortBy(state),
        total: Object.keys(state.results || {}).length
    }),
    {
        setSorting
    }
)(SubHeader)
