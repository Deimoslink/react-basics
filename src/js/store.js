import {connectRoutes} from 'redux-first-router';
import {applyMiddleware, createStore, combineReducers, compose} from 'redux';
import {createLogger} from 'redux-logger';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import reducer from './reducers'
import routesMap from './routes';

const history = createHistory();

const { reducer: locationReducer, middleware: locationMiddleware, enhancer } = connectRoutes(history, routesMap) // yes, 3 redux aspects

const middlewares = applyMiddleware(thunk, createLogger({collapsed: false}), locationMiddleware);
const store = createStore(combineReducers({ location: locationReducer, ...reducer }), compose(enhancer, middlewares));

export default store;
