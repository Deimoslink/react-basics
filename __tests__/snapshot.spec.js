import React from 'react';
import renderer from 'react-test-renderer';
import SubHeader from "../src/js/components/search/SubHeader.jsx";
import store from '../src/js/store';

const component = renderer.create(
    <SubHeader store={store}/>
).toJSON();

describe('subheader renders correctly', () => {
    it('', () => {
        expect(component).toMatchSnapshot();
    });


});


