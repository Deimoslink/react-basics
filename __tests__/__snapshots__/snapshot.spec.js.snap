// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subheader buttons change class when active  1`] = `
<div
  className="sub-header"
>
  <div>
    <span
      hidden={true}
    >
      0
       movies found
    </span>
  </div>
  <div
    className="filter-panel"
  >
    <span>
      Sort by
    </span>
    <button
      className="button button-simple null"
      onClick={[Function]}
    >
      release date
    </button>
    <button
      className="button button-simple active"
      onClick={[Function]}
    >
      rating
    </button>
  </div>
</div>
`;
